ynput/OpenPype
# Data Exchanges: Point Cloud for 3dsMax

## Brief description
Publish PRT format with tyFlow in 3dsmax

## Description
Publish PRT format with tyFlow in 3dsmax and possibly set up loader to load the format too.
- [x] creator 
- [x] extractor 
- [x] validator

## Additional info
1. Users need to manually create export_particle operator for publishing to export the point cloud in PRT format as tyFlow API currently doesn't support creating any operator in editor.
2. Partition count and related set-up are not allowed due to the current renaming from integrate,py does not support weird naming in terms of parition count. 
3. currently supports prt format only.
4. Make sure you installed tyFlow plugin, otherwises it will error out if publish the instance.
5. Point Cloud would be loaded within tyCache in Loader.

## Testing notes:
1. launch 3dsmax via Openpype
2. Click Publish from Openpype
3. Create PointCloud instance with your tyFlow class
6. Before publishing, make sure you add export_particles operator into your tyFlow editor(it can be accessed through modifier)
7. If you forgot to add export_particles, the validator will remind you and it will error out
8. If you have any custom channel attributes needed to add, you can add through setting
![image](https://user-images.githubusercontent.com/64118225/222127795-4f3c5d55-8669-4018-9612-f460b9ae5b71.png)
9. If the custom attributes doesn't exist, the validator will also error out.
10. If you manually changed the partition settings, the validator will error out too.
11. Publish Point Cloud
